core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala

```diff
@@ -285,6 +292,11 @@ private[spark] class ExecutorAllocationManager(
 
     updateAndSyncNumExecutorsTarget(now)
 
+    if (preemptableExecutors.nonEmpty) {
```

Comment:
Instead of having the preemption policy modify private fields of this class, triggering this asynchronous removal, it would be better to have the preemption policy call the existing (albeit unused) removeExecutor() method in this class.